temporal-sdk/src/main/java/io/temporal/common/converter/CodecDataConverter.java

```diff
@@ -60,14 +76,53 @@ public class CodecDataConverter implements DataConverter, PayloadCodec {
    *   <li>{@code dataConverter} is applied last
    * </ul>
    *
+   * See {@link #CodecDataConverter(DataConverter, Collection, boolean)} to enable encryption of
+   * Failure's default attributes.
+   *
    * @param dataConverter to delegate data conversion to
    * @param codecs to delegate bytes encoding/decoding to. When encoding, the codecs are applied
    *     last to first meaning the earlier encoders wrap the later ones. When decoding, the decoders
    *     are applied first to last to reverse the effect
    */
   public CodecDataConverter(DataConverter dataConverter, Collection<PayloadCodec> codecs) {
+    this(dataConverter, codecs, false);
+  }
+
+  /**
+   * When serializing to Payloads:
+   *
+   * <ul>
+   *   <li>{@code dataConverter} is applied first, following by the chain of {@code codecs}.
+   *   <li>{@code codecs} are applied last to first meaning the earlier encoders wrap the later ones
+   * </ul>
+   *
+   * When deserializing from Payloads:
+   *
+   * <ul>
+   *   <li>{@code codecs} are applied first to last to reverse the effect following by the {@code
+   *       dataConverter}
+   *   <li>{@code dataConverter} is applied last
+   * </ul>
+   *
+   * Setting {@code encodeDefaultAttributes} to true enables codec encoding of Failure's default
+   * attributes. This can be used in conjunction with an encrypting codec to enable encryption of
+   * failures message and stack traces. Note that failure's details are always codec-encoded,
+   * without regard to {@code encodeDefaultAttributes}.
+   *
+   * @param dataConverter to delegate data conversion to
+   * @param codecs to delegate bytes encoding/decoding to. When encoding, the codecs are applied
+   *     last to first meaning the earlier encoders wrap the later ones. When decoding, the decoders
+   *     are applied first to last to reverse the effect
+   * @param encodeDefaultAttributes enable encoding of Failure's default attributes (message and
```

Comment:
Let's rename to `encodeFailures`. `default attributes` is a waaaay to internal-api level of naming to be clear for users.